src/core/tour/TourManager.ttslua

```diff
@@ -0,0 +1,188 @@
+  -- @param playerColor Player color to hide the current card for
+  internal.hideCard = function(playerColor)
+    Global.UI.hide(cardId)
+  end
+
+  -- Cleans up all the various resources associated with the tour, and (hopefully) resets the
+  -- camera to the default position.  Camera handling is erratic, the final card in the script
+  -- should include instructions for the player to fix it.
+  -- @param playerColor Player color to clean up
+  internal.finalizeTour = function(playerColor)
+    local cameraHook = getObjectFromGUID(cameraHookGuid)
+    cameraHook.destruct()
+    Player["White"].setCameraMode("ThirdPerson")
+    Wait.frames(function()
+      -- This resets to the default camera position.  If we don't place the camera exactly at the
+      -- default, camera controls get weird
+      Player["White"].lookAt({position={-22.265,-2.5,5.2575},pitch=64.343,yaw=90.333,distance=104.7})
+    end, 3)
+  end
+
+  -- Updates the card UI to show the appropriate narrator and text.
+  -- @param index Script entry which should be shown
+  -- @param playerColor Player color to update card for
+  internal.updateCardDisplay = function(index, playerColor)
+    Global.UI.setAttribute(narratorId, "image", TOUR_SCRIPT[index].narrator)
+    Global.UI.setAttribute(textId, "text", TOUR_SCRIPT[index].text)
+  end
+
+  -- Creates a small, transparent object which the camera will be attached to in order to move the
+  -- user's view around the table.  This should be called only at the beginning of the tour. Once
+  -- creation is complete the user's camera will be attached to the hook and the first card will be
+  -- shown.
+  -- @param playerColor Player color to create the hook for
+  internal.createCameraHook = function(playerColor)
+    local hookData = {
+      Name = "BlockSquare",
+      Transform = {
+        posX = CAMERA_HOME.x,
+        posY = CAMERA_HOME.y,
+        posZ = CAMERA_HOME.z,
+        rotX = 0,
+        rotY = 270.0,
+        rotZ = 0,
+        scaleX = 0.1,
+        scaleY = 0.1,
+        scaleZ = 0.1,
+      },
+      ColorDiffuse = {
+        r = 0,
+        g = 0,
+        b = 0,
+        a = 0,
+      },
+      Locked = true,
+    }
+
+    spawnObjectData({ data = hookData, callback_function = internal.onHookCreated })
+  end
+
+  -- Callback for creation of the camera hook object.  Will attach the camera and show the current
+  -- (presumably first) card.
+  -- @param hook Created object
+  internal.onHookCreated = function(hook)
+    cameraHookGuid = hook.getGUID()
+    Player.White.attachCameraToObject({
+      object = hook,
+      offset = { x = -20, y = 30, z = 0 }
+    })
+    internal.showCurrentCard()
+  end
+
+  -- Creates an XMLUI entry in Global for a player-specific tour card.  Dynamically creating this
+  -- is somewhat complex, but ensures we can properly handle any player color.
+  -- @param playerColor Player color to create the card for
+  internal.createTourCard = function(playerColor)
```

Comment:
How about  a handling for playerColor = "All"?

Reply:
With the camera controls being distinct for each player an All doesn't really work here.  It might theoretically be possible to hook all the active players, but the camera controls are so touchy I'm afraid to try.  Would also be some complicated questions like what happens if a new player joined mid-tour, that just aren't worth dealing with.